packages/widgets/src/components/nk-drop-mint-button/nk-drop-mint-button.tsx

```diff
@@ -0,0 +1,211 @@
+import Swal from 'sweetalert2';
+import state from '../../stores/wallet';
+
+@Component({
+  tag: 'nk-drop-mint-button',
+  styleUrl: 'nk-drop-mint-button.scss',
+  shadow: true,
+})
+export class NKDropMintButton {
+  @State() disabled = true;
+
+  @State() loading = true;
+
+  @State() supply: number;
+
+  @State() maxAmount: number;
+
+  @State() maxPerMint: number;
+
+  @State() saleActive: boolean;
+
+  @State() presaleActive: boolean;
+
+  @State() selections: number[] = [];
+
+  @State() selectedValue = -1;
+
+  @Prop() placeholder = 'Mint NFT';
+
+  @Prop() successTitle = 'Success';
+
+  @Prop() successMessage = 'Successfully minted an NFT';
+
+  container!: HTMLDivElement;
+
+  select: MDCSelect | null = null;
+
+  selectedText: HTMLSpanElement;
+
+  disconnect: () => void;
+
+  componentWillLoad() {
+    this.disconnect = watchBlockNumber({ listen: true }, async () => {
+      this.supply = (await state.diamond.base.totalSupply()).toNumber();
+      this.maxAmount = (await state.diamond.apps.drop.maxAmount()).toNumber();
+      this.maxPerMint = (await state.diamond.apps.drop.maxPerMint()).toNumber();
+      this.saleActive = await state.diamond.apps.drop.saleActive();
+      this.presaleActive = await state.diamond.apps.drop.presaleActive();
+      this.selections = Array(this.maxPerMint).fill('');
+      this.loading = false;
+
+      // if sale is active, enable the widget
+      if (this.saleActive || this.presaleActive) {
+        this.disabled = false;
+      }
+    });
+  }
+
+  componentDidLoad() {
+    this.setSelectedText();
+  }
+
+  componentDidUpdate() {
+    if (this.selections.length > 0 && !this.select) {
+      this.select = new MDCSelect(this.container);
+      this.select.listen('MDCSelect:change', () => {
+        // trigger mint
+        this.selectedValue = Number(this.select.value);
+        this.mint(this.selectedValue);
+      });
+    }
+
+    this.setSelectedText();
+  }
+
+  disconnectedCallback() {
+    this.disconnect();
+  }
+
+  @Method()
+  async mint(quantity: number) {
+    try {
+      this.loading = true;
+      const { address } = state.client.getAccount();
+      if (this.presaleActive) {
+        const verify = await state.diamond.verify(address);
+
+        // calculate fees
+        const tx = await state.diamond.apps.drop.presaleMintTo(
+          address,
+          quantity,
+          verify.allowed,
+          verify.proof
+        );
+
+        await tx.wait();
+
+        await Swal.fire({
+          title: this.successTitle,
+          text: this.successMessage,
+        });
+
+        return;
+      }
+
+      if (this.saleActive) {
+        const tx = await state.diamond.apps.drop.mintTo(address, quantity);
+
+        await tx.wait();
+
+        await Swal.fire({
+          title: this.successTitle,
+          text: this.successMessage,
+        });
+
+        return;
+      }
+
+      throw new Error('Sale is not active.');
+    } catch (err) {
```

Comment:
we should probably have a global `handleError` function to properly format blockchain errors